overlord/devicestate/handlers_systems.go

```diff
@@ -204,7 +204,19 @@ func (m *DeviceManager) doCreateRecoverySystem(t *state.Task, _ *tomb.Tomb) (err
 		return logNewSystemSnapFile(filepath.Join(recoverySystemDir, "snapd-new-file-log"), where)
 	}
 
-	db := assertstate.DB(st)
+	var db asserts.RODatabase
+	if isRemodel {
+		// during remodel, the model assertion is not yet present in the
+		// database, hence we need to use a temporary one to which the
+		// assertion has been explicitly added
```

Comment:
...
// as createSystemForModelFromValidatedSnaps expects all relevant assertions to be in the passed db